tests/_helpers/github-workflows.js

```diff
@@ -0,0 +1,70 @@
+const GitHubClient = require( './github-client' );
+
+function verifyWorkflowStatus( workflowObject, reportCurrentStatus, waitingTime ) {
+	return new Promise( ( resolve, reject ) => {
+		if( workflowObject.status === 'completed' ) {
+			resolve( workflowObject );
+			return;
+		}
+		waitingTime = waitingTime || 1000;
+
+		reportCurrentStatus( workflowObject, waitingTime );
+
+		setTimeout( async () => {
+			const workflow = await getWorkflowRun( workflowObject.id );
+			// Give some time between rechecks
+			const result = await verifyWorkflowStatus( workflow.data, reportCurrentStatus, waitingTime + 3500 );
+			resolve( result );
+		}, waitingTime );
+	} );
```

Comment:
WDYT about making it `async` function only? It will only require wrapping `setTimeout` into a promise (like [mentioned here](https://stackoverflow.com/a/33292942/646871)) and then you can get rid of `new Promise` wrapper and go with `await` only.